explorer/src/providers/accounts/utils/metadataHelpers.ts

```diff
@@ -0,0 +1,192 @@
+import { AccountInfo, Connection, PublicKey } from "@solana/web3.js";
+import { Metadata, METADATA_SCHEMA } from "metaplex/classes";
+import { MetadataKey, METADATA_PREFIX, StringPublicKey } from "metaplex/types";
+import { deserializeUnchecked, BinaryReader, BinaryWriter } from "borsh";
+import base58 from "bs58";
+import {
+  METADATA_PROGRAM_ID,
+  SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID,
+  METAPLEX_ID,
+  BPF_UPGRADE_LOADER_ID,
+  SYSTEM,
+  MEMO_ID,
+  VAULT_ID,
+  AUCTION_ID,
+  toPublicKey,
+} from "metaplex/ids";
+import { findProgramAddress } from "metaplex/utils";
+import { reportError } from "utils/sentry";
+import { TOKEN_PROGRAM_ID } from "providers/accounts/tokens";
+import { Cluster } from "providers/cluster";
+
+let STORE: PublicKey | undefined;
+
+export const programIds = () => {
+  return {
+    token: TOKEN_PROGRAM_ID,
+    associatedToken: SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID,
+    bpf_upgrade_loader: BPF_UPGRADE_LOADER_ID,
+    system: SYSTEM,
+    metadata: METADATA_PROGRAM_ID,
+    memo: MEMO_ID,
+    vault: VAULT_ID,
+    auction: AUCTION_ID,
+    metaplex: METAPLEX_ID,
+    store: STORE,
+  };
+};
+
+export async function getMetadata(
+  pubkey: PublicKey,
+  cluster: Cluster,
+  url: string
+) {
+  try {
+    const accountInfo = await fetchMetadataFromPDA(pubkey, url);
+    if (accountInfo && accountInfo.data.length > 0) {
+      if (!isMetadataAccount(accountInfo)) return;
+
+      if (isMetadataV1Account(accountInfo)) {
+        const metadata = decodeMetadata(accountInfo.data);
+
+        if (
+          isValidHttpUrl(metadata.data.uri) &&
+          metadata.data.uri.indexOf("arweave") >= 0
+        ) {
+          return metadata;
+        }
+      }
+    }
+  } catch (error) {
+    if (cluster !== Cluster.Custom) {
+      reportError(error, { url });
+    }
+  }
+}
+
+export async function hasEdition(
```

Comment:
Why not consume the `Edition` information? This is good enough to unblock this PR. In it's current form this change doesn't use this information but I fully intend to in a future change. Lets try and get his V1 in and then expand upon it!